ndcube/ndcollection.py

```diff
@@ -0,0 +1,258 @@
+        elif isinstance(item, tuple):
+            # Ensure item is not longer than number of aligned axes
+            if len(item) > self.n_aligned_axes:
+                raise IndexError("Too many indices")
+            for i, axis_item in enumerate(item):
+                if isinstance(axis_item, int):
+                    drop_aligned_axes_indices.append(i)
+                for j, key in enumerate(self):
+                    collection_items[j][self.aligned_axes[key][i]] = axis_item
+
+        else:
+            raise TypeError(f"Unsupported slicing type: {axis_item}")
+
+        # Use indices of dropped axes determine above to update aligned_axes
+        # by removing any that have been dropped.
+        drop_aligned_axes_indices = np.array(drop_aligned_axes_indices)
+        new_aligned_axes = collection_utils._update_aligned_axes(
+                drop_aligned_axes_indices, self.aligned_axes, self._first_key)
+
+        return collection_items, new_aligned_axes
+
+    def copy(self):
+        return copy.deepcopy(self)
+
+    def setdefault(self):
+        raise NotImplementedError("NDCollection does not support setdefault.")
+
+    def popitem(self):
+        raise NotImplementedError("NDCollection does not support popitem.")
+
+    def pop(self, key):
+        """Removes the cube corresponding to the key from the collection and returns it."""
+        # Extract desired cube from collection.
+        popped_cube = super().pop(key)
+        # Delete corresponding aligned axes
+        popped_aligned_axes = self.aligned_axes.pop(key)
+        # If first key removed, update.
+        if key == self._first_key:
+            self._first_key = list(self.keys())[0]
+
+        return popped_cube
+
+    def update(self, key, data, aligned_axes):
+        """Updates existing cube within collection or adds new cube."""
+        # Sanitize aligned axes.
+        if isinstance(aligned_axes, int):
+            aligned_axes = (aligned_axes,)
+        sanitized_axes, n_sanitized_axes = collection_utils._sanitize_aligned_axes(
+                [self[self._first_key], data],
+                (self.aligned_axes[self._first_key], aligned_axes), 2)
+        # Update collection
+        super().update({key: data})
+        self.aligned_axes.update({key: sanitized_axes[-1]})
+
+    def __delitem__(self, key):
+        super().__delitem__(key)
+        self.aligned_axes.__delitem__(key)
+        if key == self._first_key:
+            self._first_key = list(self.keys())[0]
+
+    @classmethod
+    def _init_without_sanitization(self, data, keys, aligned_axes, meta=None):
+        """
+        Initialize NDCollection without sanitizing inputs.
+
+        Inputs same as NDCollection.__init__ except aligned_axes is an arg and
+        must be None or a tuple or tuples of ints.
+        Single ints or tuple of ints not supported here.
+        """
+        # Enter data into object.
+        super().__init__(zip(keys, data))
```

Comment:
@Cadair This line no longer works.  I think it's because it's now in a classmethod.  Think am doing something wrong here...

Reply:
This should still be in the `__init__`, it kinda has to be.

Reply:
So maybe go back to the ```dont_sanitize``` kwarg in the init?